### PR TITLE
InputCommon/XInput: Expose battery as input and cleanups.

### PR DESCRIPTION
**Pull request by jordan-woyak**
Moved things that didn't need to be in the header.
Replaced c-arrays with `std::array`.
Killed some `memset`/`cmp`s.
Cleaned up some comments.
Marked classes as `final`.

### CODE REVIEW

**Comment by iwubcode** [comment]
Just curious, what's the advantage of having battery as an input?  Is this so the user can expose the battery for alternative devices?

**Comment by jordan-woyak** [comment]
@iwubcode The only useful purpose is mapping it to the battery setting of an emulated Wii Remote.
Many games will then give you alerts when your battery is low and such.

**Comment by mbc07** [comment]
This is extremely useful now that PR #8417 got merged...

File: Source/Core/InputCommon/ControllerInterface/XInput/XInput.cpp [hunk 1]
```diff
@@ -10,43 +10,120 @@
...
+  Device* m_parent;
+};
+
+class Battery final : public Core::Device::Input
 {
 public:
   Battery(const ControlState* level) : m_level(*level) {}
   std::string GetName() const override { return "Battery"; }
   ControlState GetState() const override { return m_level; }
+  bool IsDetectable() override { return false; }
```
Comment (BhaaLseN) [inline Source/Core/InputCommon/ControllerInterface/XInput/XInput.cpp]: Seems weird that this method isn't `const`, but I guess thats from the base class. Any particular reason `Input::IsDetectable` isn't `const`?

Comment (jordan-woyak) [reply Source/Core/InputCommon/ControllerInterface/XInput/XInput.cpp]: It should be. It's my my todo list. :P

**Pull request merged**